sdk/digitaltwins/digitaltwins/src/digitalTwinsClient.ts

```diff
@@ -0,0 +1,898 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+/// <reference lib="esnext.asynciterable" />
+
+import {
+  TokenCredential,
+  RestResponse,
+  RequestOptionsBase,
+  OperationOptions,
+  bearerTokenAuthenticationPolicy,
+  createPipelineFromOptions,
+  generateUuid
+} from "@azure/core-http";
+import { PageSettings, PagedAsyncIterableIterator } from "@azure/core-paging";
+import { AzureDigitalTwinsAPI as GeneratedClient } from "./generated/azureDigitalTwinsAPI";
+import {
+  AzureDigitalTwinsAPIOptionalParams,
+  DigitalTwinsGetByIdResponse,
+  DigitalTwinsAddResponse,
+  DigitalTwinsUpdateOptionalParams,
+  DigitalTwinsUpdateResponse,
+  DigitalTwinsDeleteOptionalParams,
+  DigitalTwinsGetComponentResponse,
+  DigitalTwinsUpdateComponentResponse,
+  DigitalTwinsUpdateComponentOptionalParams,
+  DigitalTwinsAddRelationshipResponse,
+  DigitalTwinsAddRelationshipOptionalParams,
+  DigitalTwinsUpdateRelationshipOptionalParams,
+  DigitalTwinsUpdateRelationshipResponse,
+  DigitalTwinsDeleteRelationshipOptionalParams,
+  DigitalTwinsSendTelemetryOptionalParams,
+  DigitalTwinsSendComponentTelemetryOptionalParams,
+  DigitalTwinsListRelationshipsResponse,
+  IncomingRelationship,
+  DigitalTwinsListIncomingRelationshipsResponse,
+  ModelData,
+  DigitalTwinModelsGetByIdResponse,
+  DigitalTwinModelsGetByIdOptionalParams,
+  DigitalTwinModelsAddResponse,
+  DigitalTwinModelsAddOptionalParams,
+  DigitalTwinModelsListResponse,
+  DigitalTwinModelsListOptionalParams,
+  DigitalTwinModelsListOptions,
+  EventRoutesGetByIdResponse,
+  EventRoute,
+  EventRoutesAddOptionalParams,
+  EventRoutesListNextResponse,
+  EventRoutesListOptionalParams,
+  QueryQueryTwinsResponse,
+  QuerySpecification
+} from "./generated/models";
+
+/**
+ * Client for Azure IoT DigitalTwins API.
+ */
+export class DigitalTwinsClient {
+  /**
+   * @internal
+   * @ignore
+   * A reference to the auto-generated AzureDigitalTwinsAPI
+   */
+  private readonly client: GeneratedClient;
+
+  // private readonly client!: AzureDigitalTwinsAPI;
+
+  /**
+   * Creates an instance of AzureDigitalTwinsAPI.
+   *
+   * Example usage:
+   * ```ts
+   * const { DigitalTwinsClient, ServiceClientCredentials } = require("@azure/digitaltwins");
+   *
+   * const client = new DigitalTwinsClient(
+   *   "<endpoint>",
+   *   new DefaultAzureCredential();
+   * );
+   * ```
+   * @param {string} endpoint The endpoint of the service. If options parameter is given than this value will be ignored.
+   * @param {TokenCredential} credential Used to authenticate requests to the service.
+   * @param {AzureDigitalTwinsAPI} [options] Used to configure the service client.
+   */
+  constructor(
+    endpoint: string,
+    credential: TokenCredential,
+    options: AzureDigitalTwinsAPIOptionalParams = {}
+  ) {
```

Comment:
there are some missing things from this constructor, take a look at TextAnalytics:

https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/textanalytics/ai-text-analytics/src/textAnalyticsClient.ts#L168

Specifically you'll want to:
- Set the user agent
- Set the logger
- Add any whitelisted headers you want to log

-

Reply:
Updated.

---
In reply to: [470202271](https://github.com/Azure/azure-sdk-for-js/pull/10560#discussion_r470202271) [](ancestors = 470202271)